corehq/apps/users/bulk_download.py

```diff
@@ -163,6 +164,27 @@ def count_users_and_groups(domain, user_filters, group_memoizer):
     return users_count + groups_count
 
 
+def dump_usernames(domain, download_id, user_filters, task):
+    writer = Excel2007ExportWriter(format_as_text=True)
+    users_count = get_commcare_users_by_filters(domain, user_filters, count_only=True)
+    DownloadBase.set_progress(task, 0, users_count)
+
+    usernames = get_mobile_usernames_by_filters(domain, user_filters)
+
+    use_transfer = settings.SHARED_DRIVE_CONF.transfer_enabled
+    filename = "{}_users_{}.xlsx".format(domain, uuid.uuid4().hex)
+    file_path = get_download_file_path(use_transfer, filename)
+    writer.open(
+        header_table=[('users', [['username']])],
+        file=file_path,
+    )
+    writer.write([('users', [[username] for username in usernames])])
+    writer.close()
+
+    expose_download(use_transfer, file_path, filename, download_id, 'xlsx')
+    DownloadBase.set_progress(task, users_count, users_count)
```

Comment:
This function still shares the same boilerplate as `dump_users_and_groups`.  Perhaps you could pull out everything that doesn't pertain to the content of the export and put it in a shared function.

Reply:
Agreed, at least the query logic and file writing could each go into a helper.

Reply:
I did extract out the query bit in a following commit.
The code to write in excel file didn't seem worth extracting since the logic to make rows, headers was different. Going to give that another though.

Reply:
Updated.